analysis/analysis.py

```diff
@@ -0,0 +1,297 @@
+    result["mean"] = description.mean
+    result["median"] = stats.scoreatpercentile(data, 50)
+    result["stdev"] = math.sqrt(description.variance)
+    result["count"] = description.nobs
+    return result
+
+
+def write_summary(summary_output_file, summary):
+    with open(summary_output_file, "w", newline="") as summary_file:
+        json.dump(summary, summary_file)
+
+
+def fill_rows(gantt_rows, csv_rows, table_rows, task_brackets, stripped_content):
+    for task_bracket in task_brackets:
+        task_start_item = stripped_content[task_bracket[0]]
+        task_finish_item = stripped_content[task_bracket[1]]
+        task_body = task_start_item[2]["task"].split(":", 1)
+        task_id = task_start_item[2]["id"]
+        task_name = task_body[0].replace("<", "").strip() + "(#" + task_id + ")"
+        task_desc = task_body[1].replace(">", "").strip()
+        task_body_json = json.loads(task_desc.replace("'", '"'))
+
+        # add main task to rows
+        task_full_desc = json.dumps(task_body_json, sort_keys=True, indent=4).replace("\n", "<br>")
+        gantt_rows.append(
+            {
+                "Task": task_name,
+                "Start": task_start_item[0],
+                "Finish": task_finish_item[0],
+                "Description": task_full_desc,
+            }
+        )
+        hops = ""
+        if task_name.startswith("Transfer"):
+            transfer_from = task_body_json["from"]
+            transfer_to = task_body_json["to"]
+            hops = str(abs(transfer_from - transfer_to))
+        table_rows.append(
+            {
+                "id": task_id,
+                "name": task_name,
+                "duration": calculate_duration(task_start_item[0], task_finish_item[0]),
+                "description": task_full_desc,
+                "hops": hops,
+            }
+        )
+        csv_rows.append(
+            [task_name, "", calculate_duration(task_start_item[0], task_finish_item[0]), hops]
+        )
+
+        # Only add subtasks for leafs of the tree
+        main_task_debug_string = "{0} {1}: {2}".format(str(task_bracket), task_name, task_desc)
+        if not has_more_specific_task_bracket(task_bracket, task_brackets):
+            subtasks = list(filter(lambda t: t[2]["id"] == task_id, stripped_content))
+            print("{0} - subtasks = {1}".format(main_task_debug_string, str(len(subtasks))))
+            print("----------------------------------------------------------------")
+            append_subtask(task_name, table_rows, csv_rows, subtasks)
+        else:
+            print(main_task_debug_string)
+            print("----------------------------------------------------------------")
+
+
+def parse_args():
+    parser = argparse.ArgumentParser(description="Raiden Scenario-Player Analysis")
+    parser.add_argument(
+        "input_file",
+        nargs="*",
+        help="File name of scenario-player log file as main input, if empty, stdin is used",
+    )
+    parser.add_argument(
```

Comment:
I'd suggest to ask for an output directory instead. That way the user doesn't have to pass three options if they want to write the output somewhere else.